fastparquet/test/test_api.py

```diff
@@ -275,3 +275,13 @@ def test_index_not_in_columns(tempdir):
     assert out.index.tolist() == ['x', 'y', 'z']
     out = pf.to_pandas(columns=['b'], index=False)
     assert out.index.tolist() == [0, 1, 2]
+
+
+def test_no_index_name(tempdir):
+    df = pd.DataFrame({'__index_name_0__': ['x', 'y', 'z'],
+                       'b': [4, 5, 6]}).set_index('__index_name_0__')
```

Comment:
This should be tested without an explicit index, since that is the behavior in question.